userspace/libscap/engine/savefile/scap_savefile.c

```diff
@@ -695,6 +696,14 @@ static int32_t scap_read_proclist(scap_reader_t* r, uint32_t block_length, uint3
 			subreadsize += readsize;
 		}
 
+		// exe_from_memfd
+		if(sub_len && (subreadsize + sizeof(uint8_t)) <= sub_len)
+		{
+			readsize = r->read(r, &(tinfo.exe_from_memfd), sizeof(uint8_t));
+			CHECK_READ_SIZE_ERR(readsize, sizeof(uint8_t), error);
+			subreadsize += readsize;
+		}
+

```

Comment:
Technically this is undefined behavior (what if the value read is 2?) but we already do this with exe_upper_layer :<

Reply:
Not sure I understand what you mean. 0 is false anything else is true. These aren't bitfields (obv) because they are addressable.

Reply:
you're correct if you *assign* an int to a bool:

```
bool foo = 2; // (foo == true)
```

but *not* if you memcpy the raw bytes

Reply:
Fixed for exe_from_memfd. We should do a separate PR for fixing others.